Source/player.cpp

```diff
@@ -3464,9 +3464,13 @@ void PlayDungMsgs()
 		myPlayer.Say(HeroSpeech::IMustBeGettingClose, 40);
 		myPlayer.pDungMsgs |= DungMsgHell;
 	} else if (!setlevel && currlevel == 16 && !myPlayer._pLvlVisited[16] && (myPlayer.pDungMsgs & DungMsgDiablo) == 0) {
-		sfxdelay = 40;
-		sfxdnum = SfxID::DiabloGreeting;
-		myPlayer.pDungMsgs |= DungMsgDiablo;
+		for (auto &monster : Monsters) {
+			if (monster.type().type == MT_DIABLO && monster.hitPoints > 0) {
+				sfxdelay = 40;
+				sfxdnum = SfxID::DiabloGreeting;
+				myPlayer.pDungMsgs |= DungMsgDiablo;
+			}
+		}

```

Comment:
Just to not loop over more then we need.
```suggestion
		for (auto &monster : Monsters) {
			if (monster.type().type != MT_DIABLO) continue;
			if (monster.hitPoints > 0) {
				sfxdelay = 40;
				sfxdnum = SfxID::DiabloGreeting;
				myPlayer.pDungMsgs |= DungMsgDiablo;
			}
			break;
		}
```